src/Controls/src/Core/Element/Element.cs

```diff
@@ -291,43 +274,67 @@ public Element Parent
 				if (RealParent == value)
 					return;
 
-				OnPropertyChanging();
-
-				if (_parentOverride == null)
-					OnParentChangingCore(Parent, value);
-
-				if (RealParent != null)
+				if (value is null)
+				{
+					RealParent.RemoveLogicalChild(this);
+				}
+				else
 				{
-					((IElementDefinition)RealParent).RemoveResourcesChangedListener(OnParentResourcesChanged);
+					// TODO Add test for switching parents
+					if (RealParent is not null && RealParent != value)
+					{
+						RealParent.RemoveLogicalChild(this);
+					}
 
-					if (value != null && (RealParent is Layout || RealParent is IControlTemplated))
-						Application.Current?.FindMauiContext()?.CreateLogger<Element>()?.LogWarning($"{this} is already a child of {RealParent}. Remove {this} from {RealParent} before adding to {value}.");
+					if (!value.LogicalChildrenInternalBackingStore.Contains(this))
+						value.AddLogicalChild(this);
```

Comment:
@StephaneDelcroix are these changes ok? I'm trying to simplify and standardize the whole LogicalChildren and Parent setup so it all just operates with a consistent set of rules. 

This PR causes `ParentSetOnNullChildBindingContext` to fail

This feels like an inaccurate test to me.

![image](https://github.com/dotnet/maui/assets/5375137/dc409fac-334a-497a-860f-667485497a07)

If I've established a parent/child relationship between two `Element`'s then I feel like the BC at that point should propagate from the parent to the child. 

All this test is really validating, is that if you create a relationship from a child to a parent, that child isn't automatically added to the `LogicalChildren` and thus the BC won't propagate.

I'm hoping (maybe with incorrect assumptions) that once there's been a parent/child relationship established that the child will automatically get added into the parent's logical children. Which generally just makes everything like hot reload/live visual tree just work without having to custom connect all the relationships.

thoughts?